src/decoder/mod.rs

```diff
@@ -1132,18 +1129,43 @@ impl<R: Read + Seek> Decoder<R> {
     }
 
     fn create_reader<'r>(
-        reader: &'r mut SmartReader<R>,
-        compression_method: CompressionMethod,
+        decoder: &'r mut Self,
```

Comment:
Taking a mutable reference to the entire reader conflicts with [ongoing work to enable multiple strips/tiles to be decoded in parallel](https://github.com/image-rs/image-tiff/pull/161). Would you instead be able to pass _jpeg_tables_ and _photometric_interpretation_ in as additional arguments? To avoid cloning the entire Vec, the type on _Image::jpeg_tables_ could be changed to `Option<Arc<Vec<u8>>>`

Reply:
> To avoid cloning the entire Vec, the type on Image::jpeg_tables could be changed to Option<Arc<Vec<u8>>>

I thought about this, but having looked at the code in `JpegReader::new()` a clone was also required as the `jpeg_tables` is then extended with the data. I thought the simplest way to reduce the number of clones was to require an owned Vec to be passed to `JpegReader::new()`. I'll commit a version with this, and you can decide if I should still change over to `Option<Arc<Vec<u8>>>`.